python/mxnet/ndarray/ndarray.py

```diff
@@ -748,8 +746,10 @@ def dtype(self):
         return _DTYPE_MX_TO_NP[mx_dtype.value]
 
     @property
-    def storage_type(self):
-        return _storage_type(self.handle)
+    def stype(self):
+        if self._stype is None:
+            self._stype = _storage_type(self.handle)
```

Comment:
This function is needed at _ndarray_cls as well. After discussion, we will keep it as is.